src/main/java/services/ParseSection.java

```diff
@@ -30,27 +34,52 @@
   private static DateTimeFormatter timeParser =
       DateTimeFormat.forPattern("MM/dd/yyyy h:mma");
 
-  public static @NotNull Object parse(@NotNull String rawData)
+  public static @NotNull
+  SectionAttribute parse(@NotNull String rawData)
       throws IOException {
     logger.info("parsing raw catalog data...");
     Document doc = Jsoup.parse(rawData);
+    doc.select("a").unwrap();
+    doc.select("i").unwrap();
```

Comment:
What's the intention behind lines 42 and 43?

Reply:
So the section HTML has this section for location of the class, but it wraps in an a tag and i tag. Line 42 and 43 remove these two tags but keep the inner HTML. By doing so, we can wrap the location of the class

Reply:
Do these lines modify the document in place or create new objects? Additionally, I think the selectors on the next few lines ignore the wrapping tags.

Reply:
When I run it without unwrapping the tag, it didn't catch the location of the class. The selectors on the next few lines don't actually ignore the wrapping tags because of the way that the HTML is structured. The room/location building is wrapped inside an 'a' tag. The selectors will select div class = "section-content clearfix" but this div will be inside the 'a' tag, so it will not be able to find it. 

Based on the [documentation](https://jsoup.org/apidocs/org/jsoup/select/Elements.html#unwrap--), it removes the matched elements from the DOM and moves the children elements up into the parents. I believe it modifies in place and does not create new objects.

Reply:
ahh gotcha that's good to know.